teuthology/provision/pelagos.py

```diff
@@ -0,0 +1,162 @@
+
+import logging
+import requests
+import re
+import time
+
+from teuthology.config import config
+from teuthology.contextutil import safe_while
+from teuthology.util.compat import  HTTPError
+
+log = logging.getLogger(__name__)
+config_section = 'pelagos'
+
+# Provisioner configuration section description see in
+# docs/siteconfig.rst
+
+def enabled(warn=False):
+    """
+    Check for required Pelagos settings
+
+    :param warn: Whether or not to log a message containing unset parameters
+    :returns: True if they are present; False if they are not
+    """
+    conf = config.get(config_section, dict())
+    params = ['endpoint', 'machine_types']
+    unset = [_ for _ in params if not conf.get(_)]
```

Comment:
probably it's easier to read to use positive conditions:

```py
all_set = all(conf.get(param) for param in params)
if not all_set and warn:
  # ...
```

Reply:
I'm sorry, I do not agree, too many param words make it disastrous to read, I prefer to not apply what you request

Reply:
@tchaikov as for positive logic I agree, but as I understand it is a copy paste from original fog.py module, and if we change it here, we need to change it for fog as well, however there is more sense to create a base class for provision. Maybe we can do this in upcoming PRs?